File: app/src/main/java/ru/job4j/todolist/contentProvider/StoreCursor.java
package ru.job4j.todolist.contentProvider;

import android.database.AbstractCursor;

import java.util.List;

import ru.job4j.todolist.Task;

public class StoreCursor extends AbstractCursor {
    private final List<Task> store;

    public StoreCursor(List<Task> store) {
        this.store = store;
    }

    @Override
    public int getCount() {
        return store.size();
    }

    @Override
    public String[] getColumnNames() {
        return new String[] {"_ID", "NAME"};
    }

    @Override
    public String getString(int column) {
        Task task = store.get(getPosition());
        String value = null;
        if (column == 1) {
            value = task.getName();
        }
        return value;
    }

    @Override
    public short getShort(int column) {
        return 0;
    }

    @Override
    public int getInt(int column) {
        return 0;
    }

    @Override
    public long getLong(int column) {
        return 0;
    }

    @Override
    public float getFloat(int column) {
        return 0;
    }

    @Override
    public double getDouble(int column) {
        return 0;
    }

    @Override
    public boolean isNull(int column) {
        return false;
    }
}

File: app/src/main/java/ru/job4j/todolist/contentProvider/StoreCursor.kt
package ru.job4j.todolist.contentProvider

import android.database.AbstractCursor
import ru.job4j.todolist.Task

class StoreCursor(private val store: List<Task>) : AbstractCursor() {

    override fun getCount(): Int = store.size

    override fun getColumnNames(): Array<String> = arrayOf("_ID", "NAME")

    override fun getString(column: Int): String? =
            if(column == 1) store[position].name else null

    override fun getShort(column: Int): Short = 0

    override fun getInt(column: Int): Int = 0

    override fun getLong(column: Int): Long = 0

    override fun getFloat(column: Int): Float = 0F

    override fun getDouble(column: Int): Double = 0.0

    override fun isNull(column: Int): Boolean = false
}